src/structures/Channel.js

```diff
@@ -82,6 +83,7 @@ class Channel extends Base {
    *   .catch(console.error);
    */
   delete() {
+    if (this.type === 'group') return Promise.reject(new Error('DELETE_GROUP_DM_CHANNEL'));
```

Comment:
Override the method in the deriving class (`PartialGroupDMChannel`) and throw the error there.